[pkg/stanza] TestFindFilesWithIOErrors is failing on Windows

Looks like it's failing on every build for the last couple days:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6264054918/job/17009873661
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6263546281/job/17008181117
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6262496682/job/17004796589

```
--- FAIL: TestFindFilesWithIOErrors (1.48s)
    --- FAIL: TestFindFilesWithIOErrors/failed_pattern_should_not_affect_others (0.02s)
        finder_test.go:249: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go:249
            	Error:      	An error is expected but got nil.
            	Test:       	TestFindFilesWithIOErrors/failed_pattern_should_not_affect_others
        finder_test.go:250: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go:250
            	Error:      	Not equal: 
            	            	expected: []string{"1.log", "2.log", "dir1/1.log", "dir1/2.log"}
            	            	actual  : []string{"1.log", "2.log", "no_permission\\1.log", "no_permission\\2.log", "dir1\\1.log", "dir1\\2.log"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,6 +1,8 @@
            	            	-([]string) (len=4) {
            	            	+([]string) (len=6) {
            	            	  (string) (len=5) "1.log",
            	            	  (string) (len=5) "2.log",
            	            	- (string) (len=10) "dir1/1.log",
            	            	- (string) (len=10) "dir1/2.log"
            	            	+ (string) (len=19) "no_permission\\1.log",
            	            	+ (string) (len=19) "no_permission\\2.log",
            	            	+ (string) (len=10) "dir1\\1.log",
            	            	+ (string) (len=10) "dir1\\2.log"
            	            	 }
            	Test:       	TestFindFilesWithIOErrors/failed_pattern_should_not_affect_others
    --- FAIL: TestFindFilesWithIOErrors/partial_failure_of_FindFiles (0.00s)
        finder_test.go:249: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go:249
            	Error:      	An error is expected but got nil.
            	Test:       	TestFindFilesWithIOErrors/partial_failure_of_FindFiles
        finder_test.go:250: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go:250
            	Error:      	Not equal: 
            	            	expected: []string{"1.log", "2.log", "dir1/1.log", "dir1/2.log"}
            	            	actual  : []string{"1.log", "2.log", "dir1\\1.log", "dir1\\2.log", "no_permission\\1.log", "no_permission\\2.log"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,6 +1,8 @@
            	            	-([]string) (len=4) {
            	            	+([]string) (len=6) {
            	            	  (string) (len=5) "1.log",
            	            	  (string) (len=5) "2.log",
            	            	- (string) (len=10) "dir1/1.log",
            	            	- (string) (len=10) "dir1/2.log"
            	            	+ (string) (len=10) "dir1\\1.log",
            	            	+ (string) (len=10) "dir1\\2.log",
            	            	+ (string) (len=19) "no_permission\\1.log",
            	            	+ (string) (len=19) "no_permission\\2.log"
            	            	 }
            	Test:       	TestFindFilesWithIOErrors/partial_failure_of_FindFiles
```

--- a/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go
+++ b/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go
@@ -6,6 +6,7 @@ package finder
 import (
 	"os"
 	"path/filepath"
+	"runtime"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -193,6 +194,9 @@ func TestFindFiles(t *testing.T) {
 }
 
 func TestFindFilesWithIOErrors(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("permissions test not valid on windows")
+	}
 	cwd, err := os.Getwd()
 	require.NoError(t, err)
 	require.NoError(t, os.Chdir(t.TempDir()))

